### PR TITLE
fix: product has options create and delete api added

### PR DESCRIPTION
**Pull request by shanialy**
product has options create and delete api added

### CODE REVIEW

File: src/infrastructure/controllers/product.controller.ts [hunk 1]
```diff
@@ -79,4 +85,20 @@ export class ProductController {
   async updateImage(@Param('id') id: number, @Param('imageId') imageId: number, @UploadedFile('profile') file: Express.Multer.File) {
     return await this.productImagesUpdateUsecase.call(id, imageId, file);
   }
+
+  //ProductOptions
+  @Post('/:id/product-options')
```
Comment (ibadsiddiqui) [inline src/infrastructure/controllers/product.controller.ts]: `/:id/options`

File: src/infrastructure/controllers/product.controller.ts [hunk 2]
```diff
@@ -79,4 +85,20 @@ export class ProductController {
...
+  @UseBefore(ValidationMiddleware(ProductHasOptionsDto))
+  @Authorized()
+  @HttpCode(201)
+  async createProductOptions(@Param('id') id: number, @Body() data: ProductHasOptionsDto) {
+    return await this.productOptionsCreateUsecase.call(id, data);
+  }
+
+  @Delete('/:productId/product-option/:id')
```
Comment (ibadsiddiqui) [inline src/infrastructure/controllers/product.controller.ts]: it should be `/:id/options/:id`

**Review by ibadsiddiqui — CHANGES_REQUESTED** [review]
review change requested

File: src/infrastructure/controllers/product.controller.ts (hunk 1 above)
Comment (shanialy) [reply src/infrastructure/controllers/product.controller.ts]: Done

File: src/infrastructure/controllers/product.controller.ts (hunk 2 above)
Comment (shanialy) [reply src/infrastructure/controllers/product.controller.ts]: done

**Review by ibadsiddiqui — APPROVED** [review]

**Pull request merged**